s3/pkg/service/object.go

```diff
@@ -375,13 +375,14 @@ func (s *s3Service) GetObject(ctx context.Context, req *pb.GetObjectInput, strea
 
 		if bucket.ServerSideEncryption.SseType == "SSE" {
 			// decrypt and write
-			decErr, decBytes := utils.DecryptWithAES256(buf[0:n], bucket.ServerSideEncryption.EncryptionKey)
+			decErr, decBytes := utils.DecryptWithAES256(buf[0:left], bucket.ServerSideEncryption.EncryptionKey)
```

Comment:
Why use left instead of n here?

Reply:
i had used n, then some commit went in which does chunked download, and the variable changed to left. Decrypt stopped working, debugged and found this

Reply:
But I cannot understand why this happened.  As my understanding, n bytes is saved in buf, not left bytes.

Reply:
Will this try to decrypt whole object for every chunk? As we discussed, decryption works on whole object

Reply:
for the decryption algorithm, we need the exact bytes, same as size of the object. n in this case will be chunk size, 2048, so decryption fails

Reply:
@kumarashit yes, you are right. that is how it is working. what confused me here, is , in the S3 layer, we have this for loop, with the 'stream.send()' inside the loop. now, i figure this will send to the API layer (the caller), but there i do not see a loop

Reply:
Have you tried object which size is more than 2M?

Reply:
@sfzeng @kumarashit looks like the chunking will make some issue with SSE on large objects, thanks for the comment and discussons. I have now separated the code for sse and non-sse in GetObject. yet to test this, will do so, and add support for multi-part PUT and GET with SSE post DAITO, as planned earlier. Works ?

Reply:
> Have you tried object which size is more than 2M?

yes @sfzeng, thanks. test screen shots below
![encandnonencbuckets](https://user-images.githubusercontent.com/48434725/71501596-65b5be80-2891-11ea-94f5-1a93d46effc4.png)
![enclargefile](https://user-images.githubusercontent.com/48434725/71501600-6a7a7280-2891-11ea-99bc-c758e7010516.png)
![nonenclargefile](https://user-images.githubusercontent.com/48434725/71501603-6e0df980-2891-11ea-95c2-c254b3e20b31.png)